src/libs/antares/study/scenario-builder/rules.h

```diff
@@ -117,8 +118,10 @@ class Rules final : private Yuni::NonCopyable<Rules>
     //! Renewable (array [0..pAreaCount - 1])
     std::vector<renewableTSNumberData> renewable;
 
-    //! hydro levels
+    //! hydro initial levels
     hydroLevelsData hydroLevels;
+    //! hydro final levels
+    hydroFinalLevelsData hydroFinalLevels;
```

Comment:
Can you re-use the existing container ?
```suggestion
    //! hydro initial levels
    hydroLevelsData hydroInitialLevels;
    //! hydro final levels
    hydroLevelsData hydroFinalLevels;
```

Reply:
This is related to comments:

I see two ways to do this. 

- Create new class hydroFinalLevelsData. 
Inherit all the methods from hydroLevelsData. 
Overwrite methods "saveToINIFile" because of prefix += "hfl,"; 
and method apply, because it copies directly into matrix study.scenarioInitialHydroLevels and I need to be copied into matrix study.scenarioFinalHydroLevels.

- make these two differences function parameters.